src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -17,6 +19,7 @@ public static IEnumerable<SyntaxKind> GetReservedKeywordKinds()
         {
             for (int i = (int)SyntaxKind.BoolKeyword; i <= (int)SyntaxKind.ImplicitKeyword; i++)
             {
+                Debug.Assert(Enum.IsDefined(typeof(SyntaxKind), (SyntaxKind)i));
```

Comment:
>(SyntaxKind)i [](http://example.com/codeflow?start=64&length=13)

nit: do we need the cast here?

Reply:
As far as I remember, there is an exception if we don't case, let me double check

Reply:
Yes, we do because the underlying type of `SyntaxKind` is `ushort`:

> System.ArgumentException : Enum underlying type and the object must be same type or object must be a String. Type passed in was 'System.Int32'; the enum underlying type was 'System.UInt16'.

Reply:
Thanks